### PR TITLE
[bugfix] Fix bug and parameterize cuda samples tests

### PR DESCRIPTION
**Pull request by teojgo**

### CODE REVIEW

File: cscs-checks/prgenv/cuda/cuda_samples.py [hunk 1]
```diff
@@ -8,14 +8,18 @@
 import reframe.utility.osext as osext
 
 
-class CudaSamples(rfm.RegressionTest):
-    def __init__(self):
+@rfm.parameterized_test(['deviceQuery'], ['concurrentKernels'],
+                        ['simpleCUBLAS'], ['bandwidthTest'],
+                        ['conjugateGradientCudaGraphs'])
+class CudaSamplesTest(rfm.RegressionTest):
+    def __init__(self, test_name):
```
Comment (jjotero) [inline cscs-checks/prgenv/cuda/cuda_samples.py:15]: I'd change this to use the new parameter syntax instead
```suggestion
@rfm.simple_test
class CudaSamplesTest(rfm.RegressionTest):
    parameter('test_name', ['deviceQuery', 'concurrentKernels', 
        'simpleCUBLAS', 'bandwidthTest', 
        'conjugateGradientCudaGraphs']
    )
    def __init__(self):
```

File: cscs-checks/prgenv/cuda/cuda_samples.py [hunk 2]
```diff
@@ -8,14 +8,18 @@
...
+        self.descr = f'CUDA {test_name} test'
```
Comment (jjotero) [inline cscs-checks/prgenv/cuda/cuda_samples.py:16]: The new parameter syntax injects the parameters directly as class attributes
```suggestion
        self.descr = f'CUDA {self.test_name} test'
```

File: cscs-checks/prgenv/cuda/cuda_samples.py [hunk 3]
```diff
@@ -33,11 +37,24 @@ def __init__(self):
...
+
         self.sourcesdir = 'https://github.com/NVIDIA/cuda-samples.git'
         self.build_system = 'Make'
         self.build_system.options = [f'SMS="{self.nvidia_sm}"',
                                      f'CUDA_PATH=$CUDA_HOME']
-        self.prebuild_cmds = ['git checkout v11.0']
+        self.prebuild_cmds = [f'git checkout v11.0', f'cd Samples/{test_name}']
+        self.executable = f'Samples/{test_name}/{test_name}'
+        self.sanity_patterns = sn.assert_found(
+            output_patterns[test_name], self.stdout
```
Comment (jjotero) [inline cscs-checks/prgenv/cuda/cuda_samples.py:56]: Need to change these too
```suggestion
        self.prebuild_cmds = [f'git checkout v11.0', f'cd Samples/{self.test_name}']
        self.executable = f'Samples/{self.test_name}/{self.test_name}'
        self.sanity_patterns = sn.assert_found(
            output_patterns[self.test_name], self.stdout
```

**Review by jjotero — CHANGES_REQUESTED** [review]
Looks good! My only suggestion is that we start using the new parametrisation syntax instead.

**Review by jjotero — APPROVED** [review]
lgtm 👍

File: cscs-checks/prgenv/cuda/cuda_samples.py [hunk 4]
```diff
@@ -8,14 +8,19 @@
 import reframe.utility.osext as osext
 
 
-class CudaSamples(rfm.RegressionTest):
+@rfm.simple_test
+class CudaSamplesTest(rfm.RegressionTest):
+    parameter('test_name', ['deviceQuery', 'concurrentKernels', 'simpleCUBLAS',
```
Comment (vkarak) [inline cscs-checks/prgenv/cuda/cuda_samples.py:13]: `test_name` can be confused with the ReFrame test name. We should choose a better one.

Comment (vkarak) [reply cscs-checks/prgenv/cuda/cuda_samples.py:13]: Also I will not merge this, because the discussion of the syntax has reopened in #1699 😬

**Review by vkarak — CHANGES_REQUESTED** [review]

File: cscs-checks/prgenv/cuda/cuda_samples.py (hunk 4 above)
Comment (teojgo) [reply cscs-checks/prgenv/cuda/cuda_samples.py:13]: Do you think I should revert it to `@rfm.parameterized_test` syntax?

Comment (vkarak) [reply cscs-checks/prgenv/cuda/cuda_samples.py:13]: No, we should just wait for #1699 to be merged, unless the tests are broken also for the systems that they are valid on.

Comment (teojgo) [reply cscs-checks/prgenv/cuda/cuda_samples.py:13]: Yes, it gets a syntax error on tsa.

Comment (vkarak) [reply cscs-checks/prgenv/cuda/cuda_samples.py:13]: Yes, revert it to `parameterized_test` syntax then.

File: cscs-checks/prgenv/cuda/cuda_samples.py [hunk 5]
```diff
@@ -51,14 +50,13 @@ def __init__(self):
...
-        self.executable = f'Samples/{self.test_name}/{self.test_name}'
+        self.prebuild_cmds = [f'git checkout v11.0', f'cd Samples/{sample}']
+        self.executable = f'Samples/{sample}/{sample}'
         self.sanity_patterns = sn.assert_found(
-            output_patterns[self.test_name], self.stdout
+            output_patterns[sample], self.stdout
         )
         self.maintainers = ['JO']
-        self.tags = {'production', 'external_resosurces'}
+        self.tags = {'production', 'external_resources'}
```
Comment (vkarak) [inline cscs-checks/prgenv/cuda/cuda_samples.py:59]: We don't need the `external_resources` tag here, because we're not using the `resourcesdir`.

**Review by vkarak — CHANGES_REQUESTED** [review]

**Review by victorusu — APPROVED** [review]
`lgtm`

**Review by vkarak — APPROVED** [review]
lgtm

**Comment by vkarak** [comment]
@jenkins-cscs retry daint tsa

**Pull request merged**